main/src/main/scala/sbt/Defaults.scala

```diff
@@ -1764,24 +1760,44 @@ object Defaults extends BuildCommon {
 
   // 1. runnerSettings is added unscoped via JvmPlugin.
   // 2. In addition it's added scoped to run task.
-  lazy val runnerSettings: Seq[Setting[_]] = Seq(runnerTask, forkOptions := forkOptionsTask.value)
+  lazy val runnerSettings: Seq[Setting[_]] = {
+    val unscoped: Seq[Def.Setting[_]] =
+      Seq(runner := ClassLoaders.runner.value, forkOptions := forkOptionsTask.value)
+    inConfig(Compile)(unscoped) ++ inConfig(Test)(unscoped)
```

Comment:
`runnerSettings` is part of `configSettings`, so this shouldn't be scoped to a particular configuration.
This will break some of the plugins - https://github.com/sbt/sbt-validator/issues/35